ls1intum/ArTEMiS
# Programming Exercise Domain Commands

### Checklist

### Motivation and Context
It would help the tas/instructors to insert tasks and testCases from domain commands from the markdown editor.

### Description
The list of test cases is taken from template participation, if there is no template participation due to an error, the test case list is empty.

To make the test case list work, I had to define a new domainCommand type "MultiOptionCommand", this class can be used to construct dropdowns in the editor.
In the markdown-editor html template a class type check is done to differentiate between domainTagCommands and multiOptionCommands.

This PR also fixes the issue with the header dropdown submitting the programming-exercise forms.

### Steps for Testing

1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Select course with a programming exercise
4. Open the exercise via edit, or create/link a new programming exercise -> commands should work as expected

### Screenshots
![image](https://user-images.githubusercontent.com/28230611/57197322-30590b00-6f66-11e9-9fbe-7194aa2a39f9.png)
